stl/inc/atomic

```diff
@@ -499,25 +539,30 @@ struct _Atomic_storage<_Ty, 1> { // lock-free using 1-byte intrinsics
         char _Prev_bytes;
 
 #if _CMPXCHG_MASK_OUT_PADDING_BITS
-        _Ty _Mask;
-        _CSTD memset(_STD addressof(_Mask), 0xff, sizeof(_Ty));
-        __builtin_zero_non_value_bits(_STD addressof(_Mask));
-        const char _Mask_val = _Atomic_reinterpret_as<char>(_Mask);
-
-        for (;;) {
-#endif // _CMPXCHG_MASK_OUT_PADDING_BITS
-            _ATOMIC_CHOOSE_INTRINSIC(_Order, _Prev_bytes, _InterlockedCompareExchange8,
-                _Atomic_address_as<char>(_Storage), _Atomic_reinterpret_as<char>(_Desired), _Expected_bytes);
-            if (_Prev_bytes == _Expected_bytes) {
-                return true;
-            }
-#if _CMPXCHG_MASK_OUT_PADDING_BITS
-            if ((_Prev_bytes ^ _Expected_bytes) & _Mask_val) {
-                break;
+        if constexpr (_Cmpxchg_has_padding_bits_v<_Ty>) {
+            _Storage_for<_Ty> _Mask{_Form_mask};
+            const long long _Mask_val = _Atomic_reinterpret_as<long long>(_Mask._Ref());

```

Comment:
`long long` here appears to be incorrect, since this is for 1-byte `_Ty`. (Indeed, 2-byte below uses `short`.)

Reply:
🤦‍♂️ Sorry for murdering your poor boy @AlexGuteniev